src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs

```diff
@@ -88,6 +88,25 @@ public bool NameEquals(ReadOnlySpan<char> text)
             return Value.TextEqualsHelper(text, isPropertyName: true);
         }
 
+        /// <summary>
+        ///   Write the property into the provided writer as a named JSON object property.
+        /// </summary>
+        /// <param name="writer">The writer.</param>
+        /// <exception cref="ArgumentException">
+        ///   This <see cref="Name"/>'s length is too large to be a JSON object property.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        ///   This <see cref="Value"/>'s <see cref="JsonElement.ValueKind"/> would result in an invalid JSON.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   The parent <see cref="JsonDocument"/> has been disposed.
+        /// </exception>>
+        public void WriteTo(Utf8JsonWriter writer)
+        {
+            writer.WritePropertyName(Name);
```

Comment:
I know this is an existing bug, but apparently we didn't add argument null validation to JsonElement's write methods when Utf8JsonWriter switched from a struct to a class.

Can you add the

```C#
if (writer == null)
    throw new ArgumentNullException(nameof(writer));
//blank line here that I can't express in this window
```
preamble to each of these WriteTo methods?

Reply:
Yeah, we can do that. Need to update some test methods too since we are using `default` for UTF8JsonWriter. 

Since you mentioned it as bug, should I create separate PR to fix this for better tracking/information purpose?

Reply:
> should I create separate PR to fix this for better tracking/information purpose

Nah. The old WriteValue method had a bug, but you deleted it.  Let's make sure the new WriteTo method doesn't have the same bug :wink:.

Reply:
Done. Added